visual-dotnet/SauceLabs.Visual/VisualClient.cs

```diff
@@ -93,6 +103,70 @@ public VisualClient(WebDriver wd, Region region, string username, string accessK
                 .Build();
         }
 
+        /// <summary>
+        /// <c>FindBuildById</c> returns the build identified by <c>buildId</c>
+        /// </summary>
+        /// <param name="buildId"></param>
+        /// <returns>the matching build</returns>
+        /// <exception cref="VisualClientException">when build is not existing or has an invalid state</exception>
+        private async Task<VisualBuild> FindBuildById(string buildId)
+        {
+            var response = await _api.Build(buildId);
+            if (response.Data?.Result == null)
+            {
+                throw new VisualClientException($@"build {buildId} was not found");
+            }
+
+            var build = response.Data.Result;
+            if (build.Mode != BuildMode.Running)
+            {
+                throw new VisualClientException($"build {build.Id} is not RUNNING");
+            }
+            return new VisualBuild(build.Id, build.Url);
+        }
+
+        /// <summary>
+        /// <c>TryFindBuildByCustomId</c> returns the build identified by <c>customId</c> or null if not found
+        /// </summary>
+        /// <param name="customId"></param>
+        /// <returns>the matching build or null</returns>
+        /// <exception cref="VisualClientException">when build has an invalid state</exception>
+        private async Task<VisualBuild?> TryFindBuildByCustomId(string customId)
```

Comment:
wouldn't it be better if we rename it to
```suggestion
        private async Task<VisualBuild?> FindBuildByCustomId(string customId)
```

Reply:
I wanted to distinguish it from `FindBuildById` which throw if not found. Any suggestion to make that distinction obvious ?